angular2/src/app/home/home.component.ts

```diff
@@ -24,13 +23,12 @@ export class HomeComponent implements OnInit {
     if(this.talkJs.getCurrentUsername()) {
       this.talkJs.getSession().then(session => {
         this.session = session;
-        this.currentUsername = this.talkJs.getCurrentUsername();
       });
     }
   }
 
   startConversation(otherParticipant: string) {
-    var me = new Talk.User({id: this.currentUsername, name: this.currentUsername});
+    const me = this.talkJs.getCurrentUser();
```

Comment:
Maybe make `me` a part of this component? Just thinking out loud.

Reply:
Decided against this because a `me` as part of the component would essentially be another `currentUser`, and it could be confusing to two. Also, we wouldn't be using it anywhere other than in this function so I don't see the benefit of making it a component field. Let me know if you disagree!

Reply:
I don't disagree. Just this component will be created anew when the user "login" again anyway.